13-1-1/recent_data.R

```diff
@@ -0,0 +1,154 @@
+  mutate(GeoCode = as.character(GeoCode)) %>% 
+  # Remove the leading code in Cause of death to make match the ons data
+  mutate(`Cause of death` = substring(`Cause of death`, 5, nchar(`Cause of death`))) %>% 
+  mutate(`Cause of death` = str_to_sentence(`Cause of death`)) %>% 
+  # Reorder columns to match the form given by historical_data
+  select(Year, 
+         Series, 
+         Country, 
+         Sex, 
+         `Cause of death`, 
+         `Observation status`, 
+         `Unit multiplier`, 
+         `Units`, 
+         GeoCode, 
+         Value)
+  
+# Now we need to aggregate for Cause of death
+
+all_deaths_cleaned_agg_by_cause <- all_deaths_cleaned %>% 
+  group_by(Year, Country, Sex) %>% 
+  summarise(Value = sum(Value)) %>% 
+  # Add the columns the disaggregate df has
+  mutate(`Cause of death` = "") %>% 
+  mutate(`Observation status` = all_deaths$OBS_STATUS_NAME %>% unique()) %>% # Time of coding: "Normal Value"
+  mutate(Units = "Number") %>% # Has to be hard-coded as "Number" not in the dataframe
+  mutate(`Unit multiplier` = "") %>% 
+  mutate(GeoCode = lapply(Country, get_GeoCode)) %>% 
+  mutate(GeoCode = as.character(GeoCode)) %>% 
+  mutate(Series = "Number of deaths from exposure to forces of nature") %>% 
+  select(Year, 
+         Series, 
+         Country, 
+         Sex, 
+         `Cause of death`, 
+         `Observation status`, 
+         `Unit multiplier`, 
+         `Units`, 
+         GeoCode, 
+         Value)
+  
+  
+
+# Clean up all_mortality----
+
+all_mortality_cleaned <- all_mortality
+all_mortality_cleaned <- all_mortality_cleaned %>% 
+  rename(Sex = GENDER_NAME,
+         `Cause of death` = CAUSE_OF_DEATH_NAME,
+         Value = OBS_VALUE,
+         Country = GEOGRAPHY_NAME,
+         Year = DATE) %>%
+  mutate(`Observation status` = all_deaths$OBS_STATUS_NAME %>% unique()) %>%
+  mutate(GeoCode = "") %>% 
+  mutate(Series = "Age-standardised mortality rates per 100,000 population") %>% 
+  mutate(`Unit multiplier` = "") %>%
+  mutate(Units = "Rate per 100,000 population") %>% 
+  select(Year, 
+         Series, 
+         Country, 
+         Sex, 
+         `Cause of death`, 
+         `Observation status`, 
+         `Unit multiplier`, 
+         `Units`,
+         GeoCode, 
+         Value) %>% 
+  # Keep only the rows for natural disaster deaths
+  filter(grepl("^[X][0-9][0-9]", `Cause of death`)) %>% 
+  # Exclude any rows corresponding to data about non EW residents
+  filter(Country %in% c("England", "Wales", "England and wales")) %>% 
+  mutate(Sex = ifelse(Sex == "Total", "", Sex)) %>%
+  mutate(`Cause of death` = "") %>% 
+  mutate(GeoCode = lapply(Country, get_GeoCode)) %>% 
+  mutate(GeoCode = as.character(GeoCode)) %>% 
+  mutate(Country = ifelse(Country=="England and wales", "", Country))
+
+# Combined the number of death data with the mortality data----
+
+recent_data <- all_deaths_cleaned %>% 
+  rbind(all_deaths_cleaned_agg_by_cause) %>% 
```

Comment:
I imagine this is fine, but as you are using tidyverse for the rest it would be more consistent to use bind_rows (it does exactly the same I believe)

Reply:
Done this